tiny_dnn/core/kernels/global_avepool_op_internal.h

```diff
@@ -0,0 +1,52 @@
+/*
+    Copyright (c) 2017, Taiga Nomi
+    All rights reserved.
+
+    Use of this source code is governed by a BSD-style license that can be found
+    in the LICENSE file.
+*/
+#pragma once
+
+namespace tiny_dnn {
+namespace kernels {
+
+inline void global_avepool_op_internal(
+  const tensor_t &in_data,
+  tensor_t &out_data,
+  const core::global_avepool_params &params,
+  const bool layer_parallelize) {
+  for_i(layer_parallelize, in_data.size(), [&](int sample) {
+    const vec_t &in = in_data[sample];
+    vec_t &out      = out_data[sample];
+
+    const size_t pool_area = params.in.width_ * params.in.height_;
+    for (size_t i = 0; i < params.in.depth_; i++) {
+      for (size_t j = 0; j < pool_area; j++) {
+        out[i] += in[i * pool_area + j];
+      }
+      out[i] /= pool_area;
```

Comment:
There's such a dilemma. And probablly Global Averaging Pooling layer operation may not be the hot spot of NN computation. So it's possible that the performance gain from optimizing this layer isn't very rewardful.

So if we need to organize optimization work, writing profiling feature and benchmark tests (layerwise? opwise?) will probablly aid the work.